avm/res/synapse/workspace/main.bicep

```diff
@@ -224,7 +224,7 @@ resource workspace 'Microsoft.Synapse/workspaces@2021-06-01' = {
       : null
     sqlAdministratorLogin: sqlAdministratorLogin
     sqlAdministratorLoginPassword: !empty(sqlAdministratorLoginPassword) ? sqlAdministratorLoginPassword : null
-    workspaceRepositoryConfiguration: workspaceRepositoryConfiguration
+    workspaceRepositoryConfiguration: !empty(workspaceRepositoryConfiguration) ? workspaceRepositoryConfiguration : null
```

Comment:
One suggestion that should do the same, but is a bit simpler:
- Undo this change
- Make the parameter `workspaceRepositoryConfiguration` nullable (`param workspaceRepositoryConfiguration object?`

So if a user doesn't provide it, the value would be null and null would be passed into the deployment.

Reply:
when working locally, this may result in `Expected a value of type "object" but the provided value is of type "object | null`,  if, in the future, the workspacerepository configuration is split up with provided values and uses a user defined type. my fix prevents that. But now I tested this locally, it is save to say this is not an issue. will update the PR.

Reply:
Hey Kathedra, I'm not sure I fully grasp the split you're referring to, but even then I would think it would be possible to handle it in a way that does not provoke a warning. In any case I appreciate that you already have UDTs on your mind. Looking forward to the time when we can import those bad boys...